src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -352,6 +355,58 @@ class spell_sha_crash_lightning : public SpellScript
     size_t _targetsHit = 0;
 };
 
+// 378270 - Deeply Rooted Elements
+class spell_sha_deeply_rooted_elements : public AuraScript
+{
+    PrepareAuraScript(spell_sha_deeply_rooted_elements);
+
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellInfo({ SPELL_SHAMAN_LAVA_BURST, SPELL_SHAMAN_ASCENDANCE_ELEMENTAL, SPELL_SHAMAN_ASCENDANCE_ENHANCEMENT, SPELL_SHAMAN_ASCENDANCE_RESTORATION })
+            && ValidateSpellEffect({ { spellInfo->Id, EFFECT_0 } });
```

Comment:
Along the comment below, you need to validate each effect
```suggestion
    && ValidateSpellEffect
    ({
        { spellInfo->Id, EFFECT_1 },
        { spellInfo->Id, EFFECT_2 },
        { spellInfo->Id, EFFECT_3 }
    });
```